Revise target file related terminology (#63)
secure-systems-lab/peps: pep-0458.txt

valuable resource used by thousands, if not millions, of people.

Before the wiki attack, PyPI used MD5 hashes to tell package managers, such as
pip, whether or not a distribution file was corrupted in transit.  However, the absence
of SSL made it hard for package managers to verify transport integrity to PyPI.
It was therefore easy to launch a man-in-the-middle attack between pip and

> PyPI, and change distribution content arbitrarily.  As a result, users could be tricked into

Reviewer: ```suggestion
PyPI, and arbitrarily change the content of distributions.  As a result, users could be tricked into
```

possible through other avenues.  For example, a public mirror is trusted to
honestly mirror PyPI, but some mirrors may misbehave, whether by accident or through
malicious intervention.
Package managers such as pip are supposed to use signatures from PyPI to verify
distribution files downloaded from a public mirror [9]_, but none are known to actually
do so [10]_.  Therefore, it would be wise to add more security measures to
detect attacks from public mirrors or content delivery networks [11]_ (CDNs).

Even though official mirrors have been deprecated on PyPI [12]_, a
wide variety of other attack vectors on package managers remain [13]_.  These attacks
can crash client systems, cause obsolete distributions to be installed, or even
allow an attacker to execute arbitrary code.  In `September 2013`__, a post was
made to the Distutils mailing list showing that the latest version of pip (at
the time) was susceptible to such attacks, and how TUF could protect users
against them [14]_.  Specifically, testing was done to see how pip would
respond to these attacks with and without TUF.  Attacks tested included replay

> and freeze, arbitrary distribution, slow retrieval, and endless data.  The post

Reviewer: isn't "arbitrary packages" a name/class of attack from the relevant paper? If not several of these attack names might want to be made more descriptive to the reader

```suggestion
and freeze, arbitrary distribution files, slow retrieval, and endless data.  The post
```

Author: You're right, "A look in the mirror" calls this attack "arbitrary package", but the spec calls it "arbitrary installation", and the reference implementation "arbitrary software installation". Let's use the spec term. :)

encouraged to review TUF design principles [2]_ and SHOULD be
familiar with the TUF specification [16]_.

The following terms used in this PEP are defined in the Python Packaging
Glossary [17]_: *project*, *release*, *distribution*.

Additional terms used in this PEP are defined as follows:

* Role: TUF specifies one *root* role and multiple other roles to which the
  *root* role delegates responsibilities, directly or indirectly. The term
  *top-level* role refers to the *root* role and any role specified directly by
  the *root* role, i.e. *timestamp*, *snapshot* and *targets* roles. Each role
  has a single metadata file that it is trusted to provide.

* Distribution file: A versioned archive file that contains Python packages,
  modules, and other resource files that are used to distribute a release. In
  this document a distribution file may also be referred to with the single

>   word distribution.

Reviewer: Worth pointing out that this is equivalent to the term "Distribution Package" in the Glossary of the Python Packaging User Guide?

Author: I briefly considered that. But it seemed more confusing than helpful.

An alternative would be to keep using the words "distribution" and "package" as before and adopt PyPI's definition of "distribution package", which also says that both words can be used alone to say the same thing.

* Simple index: The HTML page that contains internal links to distribution
  files.

* Target files: As a rule of thumb, target files are all files on PyPI whose
  integrity should be guaranteed with TUF. Typically, this includes

>   distribution files, and PyPI metadata such as simple indices.

Reviewer: ```suggestion
  distribution files and PyPI metadata, such as simple indices.
```

A software update system must complete two main tasks to integrate with TUF.
First, it must add the framework to the client side of the update system. For
example, TUF MAY be integrated with the pip package manager. Thus, new versions
of pip going forward SHOULD use TUF by default to download and verify distributions
from PyPI before installing them. However, there may be unforeseen issues that
might prevent users from installing or updating distributions, including pip itself,
via TUF. Therefore, pip SHOULD provide an option e.g.,

> `--unsafely-disable-distribution-verification`, in order to work around such issues

Reviewer: I prefer we continue to refer to "package" in the option. The term "distribution" isn't used at all in Pip's `--help` output, but the term packages occurs 9 times.

Author: You have a point. But it seems weird if we almost completely ban the word "package" from this document.

   since the last period should be discarded. As a result, developers of new
   projects will need to re-register their projects.

> 3. If the target files themselves may have been tampered with, they can be

Reviewer: It feels disorientating to refer to target files and distributions in the same sentence when discussing the same thing

```suggestion
3. If the distribution files themselves may have been tampered with, they can be
```

Author: Makes sense. But I suggest to use "target files" twice then, because this sentence should also apply to non-distribution file target files, e.g. index files, right?
